source/bot.py

```diff
@@ -32,15 +32,21 @@ async def download(self, url: str, message: Message) -> io.BytesIO:
         """
         async with aiohttp.ClientSession(raise_for_status=True) as session:
             async with session.get(url) as resp:
-                size = int(resp.headers.get("Content-Length", 0))
+                size = int(resp.headers.get("Content-Length", "0"))
                 limit = message.guild.filesize_limit
                 if size > limit:
                     raise FileSizeException(size, limit)
-                response = await resp.read()
-                size = len(response)
-                if size > limit:
-                    raise FileSizeException(size, limit)
-                return io.BytesIO(response)
+
+                #
+                # Process chunks as they come in
+                #
+                resp_bytes = bytearray()
+                async for data, _ in resp.content.iter_chunks():
+                    resp_bytes.extend(data)
+                    if len(resp_bytes) > limit:
+                        raise FileSizeException(size, limit)
```

Comment:
What is `size` here? We may need to change `FileSizeException` such that `filesize` is optional?